### PR TITLE
ADNI2BIDS

### PR DESCRIPTION
**Pull request by jsampergonzalez**
Major changes to ADNI2BIDS converter:
- Same function paths_to_bids used for all modalities
- Same function get_images_pet used for all PET tracers (image selection code changed)
- Same function visits_to_timepoints used for all modalities that need a match between visits in files using different vistit naming format
- Replaced lambda functions by more readable pandas Dataframe filtering options, when possible
- Verified for empty Dataframes in more readable was as is df.empty
- Completed all docstrings in adni_to_bids module
- Fixed bug potentially letting images not passing QC to be selected (3 images) for T1
- Replaced sheetname by sheet_name parameter in pandas. read_excel functions to prevent errirs due to deprecation

### CODE REVIEW

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 1]
```diff
@@ -21,10 +21,19 @@ def replace_sequence_chars(sequence_name):
 
     """
     import re
-    return re.sub('[ /;*():]', '_', sequence_name)
+    return re.sub('[ /;*()<>:]', '_', sequence_name)
 
 
 def fill_zeros(s, length):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: Please provide short description

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 2]
```diff
@@ -180,6 +189,15 @@ def visits_to_timepoints_mrilist(subject, mri_list_subj, adnimerge_subj, modalit
 
 
 def select_image_qc(id_list, mri_qc_subj):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: Provide description of function, with details on the different arguments

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 3]
```diff
@@ -309,6 +327,17 @@ def remove_tmp_dmc_folder(bids_dir, image_id):
 
 
 def check_bids_t1(bids_path, container='anat', extension='_T1w.nii.gz', subjects=None):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 4]
```diff
@@ -336,6 +365,17 @@ def check_bids_t1(bids_path, container='anat', extension='_T1w.nii.gz', subjects
 
 def check_bids_dwi(bids_path, container='dwi', extension=('_acq-axial_dwi.bvec', '_acq-axial_dwi.bval',
                                                           '_acq-axial_dwi.nii.gz'), subjects=None):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 5]
```diff
@@ -392,6 +432,14 @@ def is_nan(value):
 
 
 def remove_fields_duplicated(bids_fields):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 6]
```diff
@@ -735,6 +788,18 @@ def create_adni_scans_files(clinic_specs_path, bids_subjs_paths, bids_ids):
 
 
 def find_image_path(images, source_dir, modality, prefix, id_field):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 7]
```diff
@@ -777,11 +842,22 @@ def find_image_path(images, source_dir, modality, prefix, id_field):
 
 
 def t1_pet_paths_to_bids(images, bids_dir, modality, mod_to_update=False):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_utils.py [hunk 8]
```diff
@@ -815,6 +891,18 @@ def init(args):
 
 
 def create_file(image, modality, total, bids_dir, mod_to_update):
+    """
+
```
Comment (arnaudmarcoux) [inline clinica/iotools/converters/adni_to_bids/adni_utils.py]: same

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_av45_fbb_pet.py [hunk 9]
```diff
@@ -97,32 +96,34 @@ def compute_av45_fbb_pet_paths(source_dir, csv_dir, dest_dir, subjs_list):
         amyloid_qc_subj = pd.concat([av45_qc_subj, amy_qc_subj], axis=0, ignore_index=True, sort=False)
 
         for visit in list(amyloid_qc_subj.VISCODE2.unique()):
+            if pd.isna(visit):
+                continue
```
[the remark this answers is not in the document]
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_av45_fbb_pet.py]: A note has been included in the developer guide

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_av45_fbb_pet.py [hunk 10]
```diff
@@ -97,32 +96,34 @@ def compute_av45_fbb_pet_paths(source_dir, csv_dir, dest_dir, subjs_list):
...
+
             amyloid_qc_visit = amyloid_qc_subj[amyloid_qc_subj.VISCODE2 == visit]
 
-            # TODO Check
             # If there are several scans for a timepoint we keep image acquired last (higher LONIUID)
             amyloid_qc_visit = amyloid_qc_visit.sort_values("LONIUID", ascending=False)
 
+            if amyloid_qc_visit.shape[0] == 0:
```
[the remark this answers is not in the document]
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_av45_fbb_pet.py]: A note has been included in the developer guide

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fdg_pet.py [hunk 11]
```diff
@@ -96,30 +93,36 @@ def compute_fdg_pet_paths(source_dir, csv_dir, dest_dir, subjs_list):
         pet_qc_subj = pd.concat([pet_qc_1go2_subj, pet_qc3_subj], axis=0, ignore_index=True, sort=False)
 
         for visit in list(pet_qc_subj.VISCODE2.unique()):
+            if pd.isna(visit):
```
[the remark this answers is not in the document]
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fdg_pet.py]: A note has been included in the developer guide

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fdg_pet.py [hunk 12]
```diff
@@ -96,30 +93,36 @@ def compute_fdg_pet_paths(source_dir, csv_dir, dest_dir, subjs_list):
...
+                continue
+
             pet_qc_visit = pet_qc_subj[pet_qc_subj.VISCODE2 == visit]
 
             # If there are several scans for a timepoint we keep image acquired last (higher LONIUID)
             pet_qc_visit = pet_qc_visit.sort_values("LONIUID", ascending=False)
 
+            if pet_qc_visit.shape[0] == 0:
+                continue
```
[the remark this answers is not in the document]
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_fdg_pet.py]: A note has been included in the developer guide

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 1 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Removed

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 2 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 3 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Removed

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 4 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Removed

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 5 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 6 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 7 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done

File: clinica/iotools/converters/adni_to_bids/adni_utils.py (hunk 8 above)
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_utils.py]: Done

File: clinica/iotools/converters/adni_to_bids/adni_modalities/adni_dwi.py [hunk 13]
```diff
@@ -264,27 +275,26 @@ def generate_subject_files(subj, images, dest_dir, mod_to_update):
...
 
                 bids_name = bids_file_name + '_acq-axial_dwi'
 
                 bids_dest_dir = path.join(ses_path, 'dwi')
 
                 if not os.path.exists(bids_dest_dir):
-                    os.mkdir(dest_dir)
-                command = 'dcm2niix -b n -z y -o ' + bids_dest_dir + ' -f ' + bids_name + ' ' + dwi_path
+                    os.makedirs(bids_dest_dir)
+                command = 'dcm2niix -b y -z y -o ' + bids_dest_dir + ' -f ' + bids_name + ' ' + dwi_path
```
[the remark this answers is not in the document]
Comment (jsampergonzalez) [reply clinica/iotools/converters/adni_to_bids/adni_modalities/adni_dwi.py]: Yes. Option -b is set to 'y' to generate JSON file.

**Comment by arnaudmarcoux** [comment]
What does this means (last sessions) ? 
Screen Shot 2020-01-15 at 16 52 50

**Comment by jsampergonzalez** [comment]
The "extra" sessions were being obtained from selecting the visit from 'VISCODE' column instead of 'VISCODE2' column, for some csv files. These files have been added to the list of cases where we select 'VISCODE2' column directly and extra sessions do not appear anymore.

**Pull request closed without merging**